cpp/src/arrow/util/io_util_test.cc

```diff
@@ -28,19 +28,19 @@ namespace internal {
 
 void AssertExists(const PlatformFilename& path) {
   bool exists = false;
-  ASSERT_OK(FileExists(path, &exists));
+  ASSERT_OK_AND_ASSIGN(exists, FileExists(path));
```

Comment:
Ditto for the rest of this file.

Reply:
In tests it helps sometimes to have clearly visible declarations.